projects/output-and-error-handling/main.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+)
+
+func main() {
+	for {
+		resp, err := http.Get("http://localhost:8080")
+		if err != nil {
+			fmt.Println("Server not working at the moment!")
+			fmt.Fprint(os.Stderr, "Server not working at the moment! ", err, "\n")
+			return
+		}
+		defer resp.Body.Close()
+
+		if resp.StatusCode == 429 {
+			// fmt.Printf("\n%#v\n", resp.Header)
+			fmt.Println("Please wait!")
+			retryAfterStr := resp.Header["Retry-After"][0]
+			// retryAfterNum, numErr := strconv.Atoi(retryAfterStr)
+			retryAfterNum, numErr := strconv.ParseFloat(retryAfterStr, 64)
+
+			var retryDuration time.Duration
+			if numErr != nil {
+				result, timeErr := time.Parse(http.TimeFormat, retryAfterStr)
+				if timeErr != nil {
+					fmt.Fprint(os.Stderr, "Retry after a while (1s): ", timeErr, "\n")
+					result = time.Now().Add(time.Second)
```

Comment:
Here you're using `time.Time` as your "default meaningful" type, and `time.Duration` as a way you could compute that if you needed in the case of error.

I'd suggest thinking the other way around - what you want to produce is a `time.Duration`, and sometimes you need to go via an intermediate `time.Time` on the way to compute it.

i.e. I'd say something more like:

```go
resultDuration := time.Second
resultTime, timeErr := time.Parse(http.TimeFormat, retryAfterStr)
if timeErr == nil {
    resultDuration = time.Until(resultTime)
}
```

What you're currently doing is, in the case of an error, using a hard-coded Duration to making a fake Time, to convert it back to a Duration later on.